lib/shopify_app/controller_concerns/sanitized_params.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module ShopifyApp
+  module SanitizedParams
+    protected
+
+    def sanitized_shop_name
+      @sanitized_shop_name ||= sanitize_shop_param(params)
+    end
+
+    def referer_sanitized_shop_name
+      return unless request.referer.present?
+
+      @referer_sanitized_shop_name ||= begin
+        referer_uri = URI(request.referer)
+        query_params = Rack::Utils.parse_query(referer_uri.query)
+
+        sanitize_shop_param(query_params.with_indifferent_access)
+      end
+    end
+
+    def sanitize_shop_param(params)
+      return unless params[:shop].present?
+      ShopifyApp::Utils.sanitize_shop_domain(params[:shop])
+    end
+
+    def sanitized_params
+      request.query_parameters.clone.tap do |query_params|
+        if params[:shop].is_a?(String)
```

Comment:
Should this check be in `sanitize_shop_param`?

Reply:
This has been like this for ~4 years (https://github.com/Shopify/shopify_app/pull/571/files) ... I'm just moving it out of `login_protection.rb` to be accessible elsewhere.